File: src/Samples/XAML_Features/Command/CommandDemoViewModel.fs
namespace OpenSilver.Samples.Showcase

open System
open System.Windows
open CommunityToolkit.Mvvm.ComponentModel
open CommunityToolkit.Mvvm.Input

type CommandDemoViewModel() as this =
    inherit ObservableObject()

    let showMessageCommand =
        RelayCommand<string>(
            (fun msg -> this.ShowMessage(msg)),
            (fun s -> not (String.IsNullOrWhiteSpace(s)))
        )

    let mutable message = "Message"

    member this.ShowMessageCommand: IRelayCommand = showMessageCommand

    member this.Message
        with get() = message
        //and set(value) =
        //    if this.SetProperty(&message, value) then
        //        showMessageCommand.NotifyCanExecuteChanged()

    member private _.ShowMessage(msg: string) =
        MessageBox.Show($"Command is executed: {msg}") |> ignore

File: src/Samples/XAML_Features/Command/CommandDemoViewModel.cs
using CommunityToolkit.Mvvm.ComponentModel;
using CommunityToolkit.Mvvm.Input;
using System.Windows;

namespace OpenSilver.Samples.Showcase;

public class CommandDemoViewModel : ObservableObject
{
    public IRelayCommand ShowMessageCommand { get; }

    private string _message = "Message";
    public string Message
    {
        get => _message;
        set
        {
            SetProperty(ref _message, value);
            ShowMessageCommand.NotifyCanExecuteChanged();
        }
    }

    public CommandDemoViewModel()
    {
        ShowMessageCommand = new RelayCommand<string>(ShowMessage, s => !string.IsNullOrWhiteSpace(s));
    }

    private void ShowMessage(string message)
    {
        MessageBox.Show($"Command is executed: {message}");
    }
}
